pkg/frontend/querymiddleware/model_extra.go

```diff
@@ -60,10 +60,27 @@ func (q *PrometheusRangeQueryRequest) WithQuery(query string) Request {
 	return &new
 }
 
-// WithQuery clones the current `PrometheusRangeQueryRequest` with new hints.
-func (q *PrometheusRangeQueryRequest) WithHints(hints *Hints) Request {
+// WithTotalQueriesHint clones the current `PrometheusRangeQueryRequest` with an
+// added Hint value for TotalQueries.
+func (q *PrometheusRangeQueryRequest) WithTotalQueriesHint(totalQueries int32) Request {
```

Comment:
The contract of this function is to clone the request, but the new implementation doesn't clone it because `Hints` are shared between old and new request.

_Same comment applies to `WithEstimatedCardinalityHint()`._